server/src/main/java/no/fractal/socket/send/AbstractMessage.java

```diff
@@ -0,0 +1,150 @@
+
+        // - id Size 2b and id
+        StreamUtils.writeInt(outputStream, typeBytes.length);
+        outputStream.write(typeBytes);
+
+        // - segmentArr size 2b and arr
+        StreamUtils.writeInt(outputStream, segmentBytes.length);
+        outputStream.write(segmentBytes);
+
+        for (Segment segment: segments){
+            segment.writeToStream(outputStream);
+        }
+    }
+
+    private JSONArray getMainSegmentMeta(){
+        JSONArray ret = new JSONArray();
+        segments.forEach(segment -> ret.put(segment.segmentMeta));
+        return ret;
+    }
+
+    private abstract static class Segment{
+        protected final Map<String,String> segmentMeta;
+
+        protected Segment(Map<String, String> segmentMeta) {
+            this.segmentMeta = segmentMeta;
+        }
+
+        Map<String, String> getSegmentMeta() {
+            return segmentMeta;
+        }
+
+        public int getBodyByteSize(){
+            return Integer.parseInt(segmentMeta.get(keys.size.name()));
+        }
+
+        public abstract void writeToStream(BufferedOutputStream outputStream) throws IOException;
+
+        protected enum keys{
+            size,
+            mime_type,
+            file_name
+        }
+    }
+
+    public static class FileSegment extends Segment{
+        public FileSegment(File file)  {
+            super(new HashMap<>());
+            this.file = file;
+
+            try {
+                this.segmentMeta.put(keys.size.name(), String.valueOf(file.length()));
+                this.segmentMeta.put(keys.mime_type.name(), String.valueOf(Files.probeContentType(file.toPath())));
+                this.segmentMeta.put(keys.file_name.name(), file.getName());
+
+            } catch (Exception e){ e.printStackTrace();}
+
+        }
+
+        private final File file;
+
+        @Override
+        public void writeToStream(BufferedOutputStream outputStream) throws IOException {
+            StreamUtils.writeFileToStream(outputStream,file);
+        }
+    }
+
+    public static class JsonSegment extends Segment{
+        private final byte[] bytes;
+
+        public JsonSegment(JSONObject jsonObject) {
+            super(new HashMap<>());
+            this.bytes = jsonObject.toString().getBytes();
+
+            this.segmentMeta.put(keys.size.name(), String.valueOf(bytes.length));
+            this.segmentMeta.put(keys.mime_type.name(), "application/json");
+
+        }
+
+        @Override
+        public void writeToStream(BufferedOutputStream outputStream) throws IOException {
+            outputStream.write(bytes);
+        }
+    }
+
+
+    public static class ByteSegment extends Segment{
+        private final byte[] blob;
+
+        public ByteSegment(byte[] blob) {
+            super(new HashMap<>());
+            this.blob = blob;
+
+            this.segmentMeta.put(keys.size.name(), String.valueOf(blob.length));
+            this.segmentMeta.put(keys.mime_type.name(), "application/bytes");
```

Comment:
application/bytes er ikke en gydlig sak.
application/octet-stream
Men uansett, så bør vel denne vær åpen for type injection?

Reply:
type injection? hva mener du med det?
sån tolkning av filtype?